vue/components/organisms/configurator/configurator.vue

```diff
@@ -0,0 +1,243 @@
+         * Ripe SDK instance will be used.
+         */
+        ripeSdk: {
+            type: Object,
+            default: null
+        }
+    },
+    data: function() {
+        return {
+            /**
+             * The frame that is currently being shown in the
+             * configurator.
+             */
+            frameData: this.frame,
+            /**
+             * Flag that controls if the initial loading process for
+             * the modal in the configurator is still running.
+             */
+            loading: true
+        };
+    },
+    watch: {
+        size(size) {
+            this.resize(size);
+        },
+        frame(frame) {
+            if (frame === this.frameData) return;
+            this.frameData = frame;
+        },
+        frameData(frame) {
+            // in case the configurator is not currently ready
+            // then avoids the operation (returns control flow)
+            if (!this.configurator || !this.configurator.ready) return;
+
+            this.configurator.changeFrame(frame, {
+                type: null,
+                duration: null
+            });
+        }
+    },
+    computed: {
+        elementDisplayed() {
+            if (!this.configurator) {
+                return false;
+            }
+            return getComputedStyle(this.configurator.element).display !== "none";
+        },
+        loaderStyle() {
+            return {
+                top: this.size ? `${this.size / 2}px` : "calc(50%)"
+            };
+        }
+    },
+    mounted: async function() {
+        this.$emit("loading");
+        await this.setupRipeSdk();
+
+        this.configurator = global.ripeSdk.bindConfigurator(this.$refs.configurator, {
+            view: this.getView(),
+            position: this.getPosition()
+        });
+
+        this.configurator.bind("changed_frame", frame => {
+            this.frameData = frame;
+
+            // only the visible instance of this component
+            // should be sending events it's considered to
+            // be the main/master one
+            if (this.elementDisplayed) {
+                this.$emit("update:frame", frame);
+            }
+        });
+
+        this.configurator.bind("loaded", () => {
+            const frame = `${this.configurator.view}-${this.configurator.position}`;
+            this.frameData = frame;
+            this.loading = false;
+            this.$emit("update:frame", frame);
+            this.$emit("loaded");
+        });
+
+        this.resize(this.size);
+    },
+    methods: {
+        /**
+         * Initializes Ripe SDK if it does not exists and
+         * configurates it with the given brand, model,
+         * version and parts.
+         */
+        async setupRipeSdk() {
+            global.ripeSdk = this.ripeSdk;
+
+            if (!global.ripeSdk) {
+                global.ripeSdk = new Ripe();
+            }
+
+            try {

```

Comment:
`loading = true`. Right now it's the same thing, but conceptually it makes sense